base/docker/scripts/status_tools.py

```diff
@@ -0,0 +1,83 @@
+from typing import List, Optional
+from prometheus_client import Gauge, Enum, Info
+import logging
+from enum import Enum as PyEnum
+from typer import Typer
+import os
+import requests
+
+logger = logging.getLogger(__name__)
+
+class WorkflowStatus(PyEnum):
+    STARTING = "starting"
+    RUNNING = "running"
+    STOPPING = "completed"
+    FAILED = "failed"
+
+class WorkFlowError(PyEnum):
+    DB_ERROR = "db_error"
+    WORKFLOW_ERROR = "workflow_error"
+    API_ERROR = "api_error"
+
+class StatusUpdater:
+    def post_update(
+            self,
+            completed: Optional[float] = None,
+            phases: Optional[List[str]] = None,
+            phase: Optional[str] = None,
+            error_message: Optional[str] = None,
+            error_code: Optional[WorkFlowError] = None,
+            status: Optional[WorkflowStatus] = None,
+            accessible: Optional[bool] = None):
+        wf_status = {
+        }
+        if completed is not None:
+            wf_status["completeness"] = float(completed) / 100.0
```

Comment:
I assume this is because of percentages.
My take on percentages (also degrees, imperial units, and timezones) is that they should be purely a display artefact (and possibly allowed in user entry), and should never be stored or used in APIs.